### ISSUE TITLE
VsCoq v2 unsupported command for Restart and Undo

### ISSUE THREAD

**Issue by Chesterhuang1999** [rocq-prover/vsrocq#770]
I just updated my VsCoq to v2.1.2 from v0.3.9 (sorry for the delay) but I found a problem that using Restart and Undo will cause an error message 'Unsupported Command'. It seems that the backtrack tactics are yet to be implemented in v2.1.2. Is that the case?

**Comment by thomas-lamiaux** [comment]
In know that some users don't care much about it, but if it is not too much work, it would be great to have soon. I am personally really using it day-to-day for tutorials, but also when I am working to easily compare how different tactics do different stuffs, or if I am writing a tactic to check it improves / does not regress etc...

**Comment by rtetley** [comment]
So I understand your use cases. I am not sure what to do after the discussion in #687. @ybertot what is your take on this ?

**Comment by RalfJung** [comment]
I like to use `Restart` when teaching Coq to students, where it is helpful to be able to show different ways of doing a proof -- or to show "see, here we are stuck, so let's start over again from scratch". I can work around that by using `Abort.` and repeating the lemma statement, but `Restart.` is a lot nicer.

Restart seems a lot less dangerous than `Reset`, so I don't quite see that this would have the same concerns as https://github.com/coq-community/vscoq/issues/687.

**Comment by yforster** [comment]
I agree with @RalfJung. I didn't even know `Reset` exists and I don't mind whether it is supported or not, but `Restart` is used in lots of teaching material for good reason. Removing `Restart` both makes teaching material worse and incurs the overhead of having to rewrite the teaching material so it stays compatible with vscoq2.

Would it be hard to get `Restart` back?
